linux/cuda_info.bsh

```diff
@@ -22,211 +23,370 @@ source "${VSI_COMMON_DIR}/linux/aliases.bsh"
+#
+# .. rubric:: Example
+#
+# .. code-block:: bash
+#
+#     $ echo "${CUDA_ARCHES[@]}"
+#     20 30 32 35 37 50 52 53 60 61 62
+#
+#   Adding the periods to the architecture version number:
+#
+#   y=()
+#   for x in ${CUDA_ARCHES[@]+"${CUDA_ARCHES[@]}"}; do
+#     y+=("${x:0:${#x}-1}.${x:${#x}-1:1}")
+#   done
+#
+#   $ echo "${y[@]}"
+#   2.0 3.0 3.2 3.5 3.7 5.0 5.2 5.3 6.0 6.1 6.2
+#
+# .. seealso::
+#   :var:`CUDA_DEPRECATED`
+#
+# .. var:: CUDA_CODES
+#
+# An array of CUDA "real" instruction sets supported by CUDA. Every version of CUDA (``nvcc``) has a set of "real" ``sm_xx`` architectures that that it can assemble native (CUDA binary) code for.
+#
+# This array contains a list of the sm architectures supported by the :var:`CUDA_VERSION` version of CUDA as an array of two digit numbers.
+#
+# .. seealso::
+#   :var:`CUDA_DEPRECATED`
+#
+# .. var:: CUDA_MINIMAL_DRIVER_VERSION
+#
+# Every version of CUDA has a minimal version of the NVIDIA graphics-card driver that must be installed in order to support that version of CUDA. This was largely undocumented until CUDA 10 came out, despite being obviously important. This variable is set to the minimum required version of the NVIDIA driver for the :var:`CUDA_VERSION` version of CUDA, as best as we've been able to determine.
+#
+# .. var:: CUDA_DEPRECATED
+#
+# Some versions of CUDA support old instruction sets, but print a deprecated warning. For those versions of CUDA, a :var:`CUDA_DEPRECATED` array is defined to list the two digit architectures that are supported yet deprecated in the :var:`CUDA_VERSION` version of CUDA.
+#
+# .. var:: CUDA_FORWARD_PTX
+#
+# PTX arch for newer CUDA cards. In situations where you are making portable fatbinaries, you should compile for every architecture. However, in order to future proof your fatbin for architectures newer than your current version of CUDA supports, you will need to compile to a pure virtual architecture using the PTX feature so that the real architecture can be JIT (Just-In-Time) compiled.
+#
+# :var:`CUDA_FORWARD_PTX` identifies the fullest-featured PTX architecture so that you can choose to add this to your builds.
+#*
+
+#*
+# .. var:: CUDA_CARDS
+#
+# The names of the CUDA cards. Exact values may vary, e.g. "Titan X (Pascal)" vs "Titan Xp".
+#
+# ..var:: CUDA_CARD_FAMILIES
+#
+#   The family name of each card in :var:`CUDA_CARDS`.
+#
+# .. var:: CUDA_CARD_ARCHES
+#
+# The specific CUDA architecture a card natively supports, for each card in :var:`CUDA_CARDS`.
+#*
+
+#*
+# .. var:: CUDA_SUGGESTED_ARCHES
+#
+# Suggested "virtual" architectures to compile for. Instead of compiling for every architecture that the :var:`CUDA_VERSION` version of CUDA supports, :var:`CUDA_SUGGESTED_ARCHES` is the intersection between :var:`CUDA_CARD_ARCHES` and :var:`CUDA_ARCHES` so that you compile only for your cards.
```

Comment:
Is it a problem that CUDA_CARD_ARCHES seems to be a number like 6.1 whereas CUDA_ARCHES is 61

Reply:
That is not the case

```
$ (. linux/cuda_info.bsh; discover_cuda_all; declare -p CUDA_CARD_ARCHES CUDA_ARCHES)
declare -a CUDA_CARD_ARCHES=([0]="61" [1]="61" [2]="30")
declare -a CUDA_ARCHES=([0]="35" [1]="37" [2]="50" [3]="52" [4]="53" [5]="60" [6]="61" [7]="62" [8]="70" [9]="72" [10]="75" [11]="80" [12]="86")
```

The unit test (and thus some of the code) is wrong

Reply:
Well, it was inconsistent, but not wrong. It's consistent now